### ISSUE TITLE
LSTM always return 1 values

### ISSUE THREAD

**Issue by vinhqdang** [fchollet/keras#1261]
Hello all

I am using LSTM (follow an example at https://github.com/fchollet/keras/blob/master/examples/imdb_bidirectional_lstm.py) to classify the text with 6 classes.

My code can be found at: 

http://pastebin.com/F6XjX6TP

However, the results when I printed the values is always 1 (my classes ran from 0 to 5), and of course, it does not work.

How should I fix the issue?

Thanks a lot,

**Comment by dandxy89** [comment]
I think you need 
```
model.add_node(Dense(nb_classes, activation='softmax'), name='softmax', input='dropout')
model.add_output(name='output', input='softmax')
```
to replace 
```
model.add_node(Dense(1, activation='softmax'), name='sigmoid', input='dropout')
model.add_output(name='output', input='sigmoid')
```

I would recommend you take a look at the first few rows of *Y_train* or *Y_test*.

**Comment by vinhqdang** [comment]
Thanks a lot, I will try.

To be honest, I do not think that there is a problem with Y_train and Y_test (I checked it before, and this data works well with random forest - of course the accuracy is not very high, but it still can predict something).

**Comment by vinhqdang** [comment]
Hi, 

Here is the Y_train data:


Y_train
array([[ 0.,  0.,  0.,  0.,  0.,  1.],
       [ 0.,  0.,  0.,  0.,  0.,  1.],
       [ 0.,  0.,  1.,  0.,  0.,  0.],
       ..., 
       [ 0.,  0.,  1.,  0.,  0.,  0.],
       [ 0.,  1.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  1.]])

**Comment by vinhqdang** [comment]
and X_test:

X_test
array([[  944,     2,   462, ...,   247,  1811,   372],
       [ 4463,  7195,   508, ...,   661, 14086,   316],
       [ 5661,   333,   374, ...,  1073,  4260,  1917],
       ..., 
       [ 3808,   360,   453, ...,   116,   791,  2305],
       [  867,   121,  7971, ...,  1036,  1036,   225],
       [    4, 14552,    13, ..., 15635, 16635,  1917]], dtype=int32)

**Comment by fchollet** [comment]
dandxy89 has it right.

You wrote:
`model.add_node(Dense(1, activation='softmax'), name='sigmoid', input='dropout')`

This will always return 1. Use `activation="sigmoid"` instead.
